src/core/lib/surface/passive_listener_internal.h

```diff
@@ -56,16 +57,20 @@ class PassiveListenerImpl final : public PassiveListener {
   absl::Status AcceptConnectedFd(GRPC_UNUSED int fd) override;
 
  private:
-  friend class grpc::ServerBuilder;
+  // note: the grpc_core::Server redundant namespace qualification is required
+  // for older gcc versions.
+  friend absl::Status(::grpc_server_add_passive_listener)(
+      grpc_core::Server* server, grpc_server_credentials* credentials,
+      PassiveListenerImpl& passive_listener);
 
   // Data members will be populated when initialized.
-  grpc_core::RefCountedPtr<grpc_core::Server> server_;
+  Server* server_ = nullptr;
```

Comment:
I think this still needs to hold a ref to the server.  Otherwise, it will crash if the application calls `AcceptConnectedFd()` after the server is destroyed.

Reply:
Alright, it's done, but I'm not super happy about adding another friend method to grpc_core::Server to access `Ref()`. I'll think on ways to change that.